pkg/webhook/webhook_test.go

```diff
@@ -728,39 +728,44 @@ func TestHandle(t *testing.T) {
 	decoder, _ := atypes.NewDecoder(runtime.NewScheme())
 
 	tests := []struct {
-		name               string
-		serviceAccountName string
-		podLabels          map[string]string
-		clientObjects      []client.Object
-		readerObjects      []client.Object
+		name          string
+		rawPod        []byte
+		clientObjects []client.Object
+		readerObjects []client.Object
 	}{
 		{
-			name:               "service account in cache",
-			serviceAccountName: "sa",
-			clientObjects:      serviceAccounts,
-			readerObjects:      nil,
+			name:          "service account in cache",
+			rawPod:        newPodRaw("pod", "ns1", "sa", nil, nil, false),
+			clientObjects: serviceAccounts,
+			readerObjects: nil,
 		},
 		{
-			name:               "service account not in cache",
-			serviceAccountName: "sa",
-			clientObjects:      nil,
-			readerObjects:      serviceAccounts,
+			name:          "service account not in cache",
+			rawPod:        newPodRaw("pod", "ns1", "sa", nil, nil, false),
+			clientObjects: nil,
+			readerObjects: serviceAccounts,
 		},
 		{
 			name:          "default service account in cache",
+			rawPod:        newPodRaw("pod", "ns1", "", nil, nil, false),
 			clientObjects: serviceAccounts,
 			readerObjects: nil,
 		},
 		{
 			name:          "default service account not in cache",
+			rawPod:        newPodRaw("pod", "ns1", "", nil, nil, false),
 			clientObjects: nil,
 			readerObjects: serviceAccounts,
 		},
 		{
-			name: "pod has the required label, no warnings",
-			podLabels: map[string]string{
-				UseWorkloadIdentityLabel: "true",
-			},
+			name:          "pod has the required label, no warnings",
+			rawPod:        newPodRaw("pod", "ns1", "sa", map[string]string{UseWorkloadIdentityLabel: "true"}, nil, false),
+			clientObjects: serviceAccounts,
+			readerObjects: nil,
+		},
+		{
+			name:          "pod has the required label, restart policy in init container",
```

Comment:
Also add tests where we fabricate fields that will never exist at different nesting levels that don't exist anywhere in `corev1.Pod`.  Like `pod.Panda` or `pod.Spec.Panda` or `pod.Spec.Jungle.Panda`.